storage/src/commit_log/mod.rs

```diff
@@ -0,0 +1,430 @@
+// Copyright (c) SimpleStaking and Tezedge Contributors
+// SPDX-License-Identifier: MIT
+
+mod compression;
+
+use failure::Fail;
+use serde::{Deserialize, Serialize};
+
+use crate::persistent::codec::{Decoder, Encoder, SchemaError};
+use crate::persistent::schema::{CommitLogDescriptor, CommitLogSchema};
+use crate::persistent::BincodeEncoded;
+use std::{fmt, io};
+
+use std::path::{Path, PathBuf};
+use std::sync::{Arc, RwLock};
+
+use crate::commit_log::compression::{zstd_compress, zstd_decompress};
+use std::collections::HashMap;
+use std::fs::{File, OpenOptions};
+use std::io::{BufReader, BufWriter, Read, Seek, SeekFrom, Write};
+
+pub type CommitLogRef = Arc<RwLock<CommitLog>>;
+
+const DATA_FILE_NAME: &str = "table.data";
+
+pub struct CommitLog {
+    data_file: File,
+    data_file_path: PathBuf,
+    use_compression: bool,
+}
+
+impl CommitLog {
+    pub fn new<P: AsRef<Path>>(dir: P, use_compression: bool) -> Result<Self, CommitLogError> {
+        if !dir.as_ref().exists() {
+            std::fs::create_dir_all(dir.as_ref())?;
+        }
+        let mut data_file_path = PathBuf::new();
```

Comment:
@mambisi 
maybe, this could be simplified like: `dir.as_ref().join(DATA_FILE_NAME)`

Reply:
Ok